test/test_RTUMaster.cpp

```diff
@@ -174,15 +185,16 @@ TEST_F(RTUMasterTest, it_accounts_for_invalid_data) {
     ASSERT_EQ(9, driver.getStatus().bad_rx);
 }
 
-TEST_F(RTUMasterTest, it_does_timeout_if_the_CRC_error_is_permanent) {
+TEST_F(RTUMasterTest, it_throws_if_the_error_count_reaches_the_max_value) {
```

Comment:
You are not testing that it is retrying (which was a bug of the original test).

And you now need to test for other problems, like the length problem *which again is what we are trying to solve here*.